client-js/app/elements/labrad-plot.ts

```diff
@@ -654,56 +685,185 @@ export class Plot extends polymer.Base {
 
 
   /**
-   * Generate the geometries and materials for a 1D plot.
+   * Helper method to get the scene object, or create it if it doesn't already
+   * exist.
    */
-  private plotData1D(data: number[][]): void {
-    this.dataLimits1D(data);
-
-    if (this.sceneObjects.length == 0) {
+  private getSceneObject() {
+    if (this.sceneObjects.length === 0) {
       const ob = new THREE.Object3D();
       this.sceneObjects.push(ob);
       this.scene.add(ob);
     }
 
-    const ob = this.sceneObjects[0];
+    return this.sceneObjects[0];
+  }
+
+
+  /**
+   * Plots a line given a set of data, adding it to the scene.
+   */
+  private plotLine(data: number[][], yColumn: number = 1,
+                   color: string = "#000", lineWidth: number = 1,
+                   lines: THREE.Line[]) {
+    if (data.length < 2) {
+      return;
+    }
+
+    const dataLength = data.length;
+
+    // The positions array is initialized in the geometry here, but positions
+    // are set withing projectGraphPositions. See plotData2D for more info.
+    const positions = new Float32Array(dataLength * 3);
+
+    // Raw data is stored inside the geometry for more efficient
+    // reprojection.
+    const dataPoints = new Float64Array(dataLength * 3);
+
+    let offset = 0;
+    for (let row of data) {
+      dataPoints[offset] = row[0];
+      dataPoints[offset + 1] = row[yColumn];
+      offset += 3;
+    }
+
+    const geometry = new THREE.BufferGeometry();
+    geometry.addAttribute('position', new THREE.BufferAttribute(positions, 3));
+    geometry.addAttribute('data', new THREE.BufferAttribute(dataPoints, 3));
 
-    for (let i of this.displayTraces) {
-      const length = (this.lastData) ? data.length + 1 : data.length;
+    const material = new THREE.LineBasicMaterial({
+      color: color,
+      linewidth: lineWidth
+    });
 
-      // The positions array is initialized in the geometry here, but positions
-      // are set withing projectGraphPositions. See plotData2D for more info.
-      const positions = new Float32Array(length * 3);
+    const line = new THREE.Line(geometry, material);
+    lines.push(line);
+  }
 
-      // Raw data is stored inside the geometry for more efficient
-      // reprojection.
-      const dataPoints = new Float64Array(length * 3);
-      let offset = 0;
+  /**
+   * Returns a dictionary where each trace (one-indexed) has been filtered to
+   * just include points that are within the currently selected fit area.
```

Comment:
This returns an array instead of a dictionary. Also, would be helpful to describe the shape of the output array a bit more; just saying "one-indexed" is a bit unclear.